enterpriseknowledgegraph/entity_reconciliation/cancel_entity_reconciliation_job_sample_test.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+
+from google.api_core.exceptions import InvalidArgument, NotFound
+
+import cancel_entity_reconciliation_job_sample
+
+project_id = os.environ["GOOGLE_CLOUD_PROJECT"]
+location = "global"
+job_id = "5285051433452986163"
+
+
+def test_cancel_entity_reconciliation_job(capsys):
+    try:
+        cancel_entity_reconciliation_job_sample.cancel_entity_reconciliation_job_sample(
+            project_id=project_id, location=location, job_id=job_id
+        )
+    except (InvalidArgument, NotFound) as e:
```

Comment:
I'm worried about swallowing the error here. I would remove the `try/except` and let the sample fail if there's an `InvalidArgument` or `NotFound`.

Reply:
Hmm, I see. The most common, general practice I've seen is either: a) create the resource during test setup and then destroy it in teardown; or b) have a persistent resource that you call the same operation on for each test run.

If I understand the code as written, given that any specific job can only be cancelled once, the test will only succeed the first time ever run and then will fail each time afterwards. That doesn't help us to monitor the health of this sample.

I think that the best option here would be to create a new job each time. If we need to increase the quota for this API, then we will do so. It would be nice to check for the number of running jobs and then create a mutex ... but that's not really an option here. We could use the strategy of setting these samples so that the tests aren't run on every pull request but only during nightly runs.